### DISCUSSION TITLE
Choose if a Task in obsidian moves to the project file if moved in Tick tick

### DISCUSSION THREAD

**Discussion by MassimoBaldrighi** [thesamim/TickTickSync#105] [Q&A] [answered]
When creating a task in obsidian and then moving it in Tick Tick to a specified folder/project I would like that the task in Obsidian stays where it is and not moving to the file of the related project in my Vault.

Briefly my use case:
During meetings I write some tasks and I need them to sync with tick tick, but I need them to stay inside the meeting note where they are created (with the related link). Then in Tick Tick I like to move them in the folder related to a project. Unfortunately when I do this in Tick Tick and go back to obsidian the task are moved to the file of the related project and there's no trace of where they are from.

Can I disable this behaviour?
(ps: thanks for the amazing plugin)

**Comment by thesamim** [comment]
In TickTick, after the task is moved, it should have a link to the the Obsidian Note it has moved to. Is that not happening for you?

I know that doesn't address your use case, but I want to make sure we don't have a bug.

**Reply by MassimoBaldrighi** [reply]
Yes this happened. I don't know why, but after the last update my obsidian is unusable and I cannot access tick tick anymore... By looking directly at the obsidian vault folder there are some task that have thousands of tick tick link... (I have some .md file with only a couple of task that now have the size of around 30MB due to this).

**Comment by thesamim** [comment, marked as answer]
> Yes this happened. I don't know why, but after the last update my obsidian is unusable and I cannot access tick tick anymore... By looking directly at the obsidian vault folder there are some task that have thousands of tick tick link... (I have some .md file with only a couple of task that now have the size of around 30MB due to this).

Did you turn on Full Vault Sync? If you did, I'm sorry but there was a bug in version prior to 1.0.23 **(fixed in 1.0.25)** that caused the #ticktick tag to be added continuously.

If you have Notepad++, or similar editor you can do a regex replace for the links. Like so:
![image](https://github.com/thesamim/TickTickSync/assets/5315625/ad645cb6-908d-4814-a7d2-b787386596d8)
click on "replace in files", repeat until it returns a "no files changed" message.

And then to get rid of the repeated #ticktick tags:
![image](https://github.com/thesamim/TickTickSync/assets/5315625/145a530f-6b45-4f73-b86c-adced7fa329c)
click on "replace in files", repeat until it returns a "no files changed" message.

**Reply by MassimoBaldrighi** [reply]
Thanks, I managed to remove those task directly from the project list in obsidian and fortunately they synced on tick tick, so now I can access it!
Thanks fore the quick and exhaustive reply.
PS: The initial question to this thread is in the end not possible right?

**Comment by thesamim** [comment]
You're most welcome.

> PS: The initial question to this thread is in the end not possible right?

I'm still thinking through the architectural implications. I will update here ASAP.

**Comment by pmgrove** [comment]
Would also love to see this feature added. I want my tasks to remain in the note where they are created regardless of where they are located in TickTick. Hope to see this soon 🤞

**Comment by thesamim** [comment]
@MassimoBaldrighi @pmgrove : 
Frankly this fell off my radar. 

With the current architecture, it is not possible IF the files have a default project/list associated with them. Long story short: Task management relies on a direct relationship between a file and a list. Tasks that aren't in a file that correlates to a project/list end up in the Inbox or the "Default Folder" set in settings. 

Which does mean that there IS a workaround here: If your tasks end up files that do not have a "default Project" associated with them, those tasks will end up in either the Inbox or the "Default Project" you set in settings. eg: if your meeting notes are associated with a project currently, you can use the command palette (Ctrl/Cmd P in Obsidian) to unset the default project/list and tasks will go to the Inbox or Default Project.

Please let me know if this needs clarification.

Sometime in the distant future, I have an architectural redesign in mind that would not rely on this relationship. But that's a long ways away.

**Reply by chrisweis** [reply]
@thesamim Thanks for guiding me to this thread, I'd missed it while searching before submitting such a similar request.  Really appreciate you investing your time and energy into this cool plugin!

**Reply by rblundon** [reply]
@thesamim In my vault, when TTS syncs the task from my daily note, it adds the link and taskid as it should.  It then puts a copy in my inbox page in my default folder location.  Now on the next sync, it breaks because it sees two tasks with the same taskid.  I "think" an easy solution would be to be able to exclude the "Default folder location" from the search in settings?  (I do this with the Tasks Plugin as well) This would allow the task to stay in the note it was created in, and let us move it where we'd like in TickTick while still solving the duplicate taskid error.

**Comment by pmgrove** [comment]
Hey @MassimoBaldrighi, thanks for the reply. TBH, a workaround for me would be to have a one-way sync from Obsidian --> TickTick. So just the task creation part when i add #ticktick to a task in Obsidian. I don't really need anything coming back from TickTick. If I fork the repo, is there a way to turn off the TickTick -> Obsidian data flow? Cheers.

**Reply by thesamim** [reply]
> Hey @MassimoBaldrighi, thanks for the reply. TBH, a workaround for me would be to have a one-way sync from Obsidian --> TickTick. So just the task creation part when i add #ticktick to a task in Obsidian. I don't really need anything coming back from TickTick. If I fork the repo, is there a way to turn off the TickTick -> Obsidian data flow? Cheers.

Not Massimo, but @thesamim :) How brave you feeling?

I absolutely make no guarantees that this will work. 

I absolutely make no guarantees that this will not mess up your vault or your TickTick.

**Kids don't try this at home.**

Ok, now that the disclaimers are out of the way:
start here: https://github.com/thesamim/TickTickSync/blob/c24e9bc1b7d63ee803db1b9851f5a1c7fcc156d2/src/syncModule.ts#L947

Pick what you want to save/updated/delete by examining how the function builds the various arrays for processing. Comment out/Delete anything you don't want. 

Good luck.

**Reply by pmgrove** [reply]
Dang! spot the GitHub noob... sorry @MassimoBaldrighi 😬 

Thanks for the pointers @thesamim, let me give it a try and see what happens 💥

**Comment by zyxxelwuts** [comment]
Thanks for the plugin! I would love to see that feature as well. For my use case, I have projects (=files) in Obsidian where I keep my tasks related to that project. Since projects have a defined lifespan, mapping a project to a ticktick folder that is permanent, does not make a lot of sense. So I would propose to have the option to ignore ticktick folder locations. The link to the note is sufficient. When syncing a task initially, of course some some default folder in the other app needs to be selected.
